src/probe_modules/module_quic_chlo.c

```diff
@@ -0,0 +1,428 @@
+	num_ports = conf->source_port_last - conf->source_port_first + 1;
+	
+	char port[16];
+	sprintf(port, "%d", conf->target_port);
+	// answers have the target port as source
+	memcpy(filter_rule, "udp src port \0", 14);
+
+	module_quic_chlo.pcap_filter = strncat(filter_rule, port, 16);
+	module_quic_chlo.pcap_snaplen = sizeof(struct ether_header) + sizeof(struct ip) + sizeof(struct udphdr) + QUIC_HDR_LEN_HASH + STREAM_FRAME_LEN + INCHOATE_CHLO_LEN + CLIENTHELLO_MIN_SIZE;
+	
+	connection_id = MakeQuicTag('S', 'C', 'A', 'N')  | ((uint64_t)MakeQuicTag('N', 'I', 'N', 'G')) << 32;
+	checker_bitmap = pbm_init();
+	return EXIT_SUCCESS;
+}
+
+int chlo_quic_global_cleanup(__attribute__((unused)) struct state_conf *zconf,
+		__attribute__((unused)) struct state_send *zsend,
+		__attribute__((unused)) struct state_recv *zrecv)
+{
+
+	return EXIT_SUCCESS;
+}
+
+int chlo_quic_init_perthread(void* buf, macaddr_t *src,
+		macaddr_t *gw, __attribute__((unused)) port_h_t dst_port,\
+		__attribute__((unused)) void **arg_ptr)
+{
+	int udp_send_msg_len = QUIC_HDR_LEN_HASH + STREAM_FRAME_LEN + INCHOATE_CHLO_LEN + CLIENTHELLO_MIN_SIZE;
+    //log_debug("prepare", "UDP PAYLOAD LEN: %d", udp_send_msg_len);
+
+	memset(buf, 0, MAX_PACKET_SIZE);
+	struct ether_header *eth_header = (struct ether_header *) buf;
+	make_eth_header(eth_header, src, gw);
+	struct ip *ip_header = (struct ip*)(&eth_header[1]);
+	uint16_t len = htons(sizeof(struct ip) + sizeof(struct udphdr) + udp_send_msg_len);
+	//log_debug("prepare", "IP LEN IN HEX %h", len);
+	make_ip_header(ip_header, IPPROTO_UDP, len);
+
+	struct udphdr *udp_header = (struct udphdr*)(&ip_header[1]);
+	len = sizeof(struct udphdr) + udp_send_msg_len;
+	make_udp_header(udp_header, zconf.target_port, len);
+
+	char* payload = (char*)(&udp_header[1]);
+
+	module_quic_chlo.packet_length = sizeof(struct ether_header) + sizeof(struct ip)
+				+ sizeof(struct udphdr) + udp_send_msg_len;
+	assert(module_quic_chlo.packet_length <= MAX_PACKET_SIZE);
+	memset(payload, 0, udp_send_msg_len);
+
+	
+	// Seed our random number generator with the global generator
+	/*
+     uint32_t seed = aesrand_getword(zconf.aes);
+	aesrand_t *aes = aesrand_init_from_seed(seed);
+	*arg_ptr = aes;
+     */
+	return EXIT_SUCCESS;
+}
+
+
+// then use this to update the hash
+__uint128_t fnv1a_128_inc(__uint128_t hash, const uint8_t* data, size_t len) {
+	__uint128_t FNV_primeHI = 16777216;
+	__uint128_t FNV_primeLO = 315;
+	__uint128_t FNV_prime = FNV_primeHI << 64 | FNV_primeLO;
+	
+	
+	for(size_t i = 0; i < len; i++) {
+		hash = hash ^ (__uint128_t)data[i];
+		hash = hash * FNV_prime;
+	}
+	
+	return hash;
+}
+
+// start with this
+__uint128_t fnv1a_128(const uint8_t* data, size_t len) {
+	__uint128_t hashHI = 7809847782465536322;
+	__uint128_t hashLO = 7113472399480571277;
+	__uint128_t hash = hashHI << 64 | hashLO;
+	return fnv1a_128_inc(hash, data, len);
+}
+
+void serializeHash(__uint128_t hash, uint8_t out_hash[12]) {
+	// endianess I don't care....
```

Comment:
If the server is checking this, the endianness is important, no? And if not, then why bother calculating it in the first place?